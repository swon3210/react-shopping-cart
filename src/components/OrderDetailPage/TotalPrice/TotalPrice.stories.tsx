import TotalPrice, { Props } from './TotalPrice';

export default {
  title: 'Components/OrderDetailPage/TotalPrice',
  component: TotalPrice,
  argTypes: {},
};

const Template = (args: Props) => <TotalPrice {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  title: '결제금액',
  priceLabel: '총 결제금액',
  price: '18,900원',
};
