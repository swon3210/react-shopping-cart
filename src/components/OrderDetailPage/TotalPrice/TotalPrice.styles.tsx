import styled from 'styled-components';

export const TotalPrice = styled.div`
  width: 496px;
  display: inline-block;
  color: ${({ theme }) => theme.TEXT_COLOR};
`;

export const Title = styled.div`
  padding-bottom: 24px;
  font-weight: bold;
  font-size: 28px;
  line-height: 28px;
  letter-spacing: 0.5px;
  border-bottom: 3px solid ${({ theme }) => theme.GRAY_200};
`;

export const Container = styled.div`
  padding-top: 73px;
`;

export const PriceWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const PriceLabel = styled.span`
  position: relative;
  font-size: 24px;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-weight: bold;
  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 6px;
    background-color: ${({ theme }) => theme.PRIMARY_COLOR};
    opacity: 0.4;
  }
`;

export const Price = styled.span`
  position: relative;
  font-size: 24px;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-weight: bold;
  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 6px;
    background-color: ${({ theme }) => theme.PRIMARY_COLOR};
    opacity: 0.4;
  }
`;
