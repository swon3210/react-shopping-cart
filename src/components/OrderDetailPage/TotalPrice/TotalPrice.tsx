import * as Styled from './TotalPrice.styles';

export interface Props {
  title: string;
  priceLabel: string;
  price: string;
}

const TotalPrice = ({ title, priceLabel, price }: Props) => {
  return (
    <Styled.TotalPrice>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Container>
        <Styled.PriceWrapper>
          <Styled.PriceLabel>{priceLabel}</Styled.PriceLabel>
          <Styled.Price>{price}</Styled.Price>
        </Styled.PriceWrapper>
      </Styled.Container>
    </Styled.TotalPrice>
  );
};

export default TotalPrice;
