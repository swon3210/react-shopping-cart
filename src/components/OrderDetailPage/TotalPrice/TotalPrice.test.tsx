import TotalPrice from './TotalPrice';
import { render } from '@testing-library/react';

describe('TotalPrice Component', () => {
  it('TotalPrice Snapshot', () => {
    const totalPriceUtil = render(<TotalPrice priceLabel="가격" title="제목" price="1000" />);
    expect(totalPriceUtil.asFragment()).toMatchSnapshot();
  });
});
