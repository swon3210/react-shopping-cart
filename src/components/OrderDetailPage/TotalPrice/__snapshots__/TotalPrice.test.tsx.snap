// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPrice Component TotalPrice Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa Srgyh"
  >
    <div
      class="sc-gsDJrp jMUdLb"
    >
      제목
    </div>
    <div
      class="sc-dkPtyc iCfvgp"
    >
      <div
        class="sc-hKwCoD jyiuQM"
      >
        <span
          class="sc-eCImvq kuebXo"
        >
          가격
        </span>
        <span
          class="sc-jRQAMF DVdTG"
        >
          1000
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
